abdm_integrator/utils.py

```diff
@@ -115,3 +117,40 @@ class APIResultsSetPagination(PageNumberPagination):
     page_size = 10
     page_size_query_param = 'page_size'
     max_page_size = 1000
+
+
+class ABDMCache:
+    """Wrapper class around Django's default cache for ABDM related cache operations"""
+    _cache = cache
+    prefix = 'abdm_'
+
+    @classmethod
+    def set(cls, key, value, timeout):
+        key = cls._key_with_prefix(key)
+        return cls._cache.set(key, value, timeout)
+
+    @classmethod
+    def get(cls, key):
+        key = cls._key_with_prefix(key)
+        return cls._cache.get(key)
+
+    @classmethod
+    def delete(cls, key):
+        key = cls._key_with_prefix(key)
+        return cls._cache.delete(key)
+
+    @classmethod
+    def _key_with_prefix(cls, key):
+        return f'{cls.prefix}{key}'
+
+
+def poll_for_data_in_cache(cache_key, total_attempts=30, interval=2):
```

Comment:
The method name hides the side effect that this actually deletes the cache key.

Reply:
Renamed to a relatively better name `poll_and_pop_data_from_cache` in 48ca9a971678044293aab265ad9e39bb84536d61